datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -1484,6 +1527,84 @@ impl Hash for ExprWrapper {
     }
 }
 
+/// Calculates the union (in the sense of `UnionExec`) `EquivalenceProperties`
+/// of  `lhs` and `rhs` according to the schema of `lhs`.
+fn calculate_union_binary(
+    lhs: EquivalenceProperties,
+    mut rhs: EquivalenceProperties,
+) -> Result<EquivalenceProperties> {
+    // TODO: In some cases, we should be able to preserve some equivalence
```

Comment:
I know this comment is copied from the old code, so no need to change it. 

however, it would be nice if we could give some hints about what could be preserved (I think there is already some non trivial support for preserving equivalence classes)

Reply:
We plan to spend some time to fully complete this TODO item in the next couple months or so 🚀